src/main/java/com/google/step/datamanager/DealManagerDatastore.java

```diff
@@ -99,54 +139,168 @@ public Deal updateDeal(Deal deal) {
     return transformEntitytoDeal(dealEntity);
   }
 
+  /**
+   * Returns a Deal object transformed from a deal entity.
+   *
+   * @param dealEntity Deal entity.
+   * @return a Deal object transformed from the entity.
+   */
+  private Deal transformEntitytoDeal(Entity dealEntity) {
+    long id = dealEntity.getKey().getId();
+    String description = (String) dealEntity.getProperty("description");
+    String photoBlobkey = (String) dealEntity.getProperty("photoBlobkey");
+    String start = (String) dealEntity.getProperty("start");
+    String end = (String) dealEntity.getProperty("end");
+    String source = (String) dealEntity.getProperty("source");
+    long posterId = (long) dealEntity.getProperty("posterId");
+    long restaurantId = (long) dealEntity.getProperty("restaurantId");
+    String timestamp = (String) dealEntity.getProperty("timestamp");
+    return new Deal(
+        id, description, photoBlobkey, start, end, source, posterId, restaurantId, timestamp);
+  }
+
+  /** Retrieves deals posted by _ followed by user */
+  private List<Deal> getDealsPublishedByFollowedRestaurantsOrUsers(
```

Comment:
why is this method `...RestaurantsOrUsers` ? Why only restaurant and user? Why is tags separate?
Maybe split all up into 3 different methods, or combine all

Reply:
The tags is kind of different since theres no need to search the datastore since there is the getDealIdsWithTag method  in dealTagManager but for restaurants/users, theres a need to search the datastore for the deal with the restaurantid/posterid value